src/mono/mono.proj

```diff
@@ -161,6 +163,50 @@
           IgnoreStandardErrorWarningFormat="true" />
   </Target>
 
+  <!-- Sets up WASI SDK if you don't have the WASI_SDK_PATH env variable set -->
+  <Target Name="ProvisionWasiSdk"
+          Condition="'$(ShouldProvisionWasiSdk)' == 'true' and '$(SkipMonoCrossJitConfigure)' != 'true'">
+    <ReadLinesFromFile File="$(MSBuildThisFileDirectory)/wasi/wasi-sdk-version.txt">
+      <Output TaskParameter="Lines" ItemName="_VersionLines" />
+    </ReadLinesFromFile>
+
+    <PropertyGroup>
+      <WASI_SDK_PATH>$(ProvisionWasiSdkDir)</WASI_SDK_PATH>
+      <WasiLocalPath>$([MSBuild]::NormalizeDirectory('$(MSBuildThisFileDirectory)', 'wasi'))</WasiLocalPath>
+      <WasiSdkVersion>%(_VersionLines.Identity)</WasiSdkVersion>
+      <WasiSdkUrl>https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-linux.tar.gz</WasiSdkUrl>
+      <WasiSdkUrl Condition="'$(HostOS)' == 'osx'" >https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-macos.tar.gz</WasiSdkUrl>
+      <WasiSdkUrl Condition="'$(HostOS)' == 'windows'" >https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-mingw.tar.gz</WasiSdkUrl>
```

Comment:
This should be included in general provisioning.

/cc @akoeplinger

Reply:
not a blocker, just raising that for tracking.

Reply:
I'll add changes for this from my branch to here, which moves this to `eng/testing/wasi-provisioning.targets`. And this can move to being provisioned by a common "system" whenever that comes about.

Reply:
https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/745